lib/stacks/api/datasets.ts

```diff
@@ -122,6 +122,7 @@ export class DatasetsApiStack extends Stack {
                 restApi,
                 props.authorizer,
                 props.applicationRole,
```

Comment:
did you mean to replace this one?

Reply:
Not sure what you mean, but this does look right.

Reply:
Admittedly I was confused at first too. The first of these two parameters looks to be a real `IRole` (sometimes it is the application role but sometimes not) and then the second is a string of the role name.